src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerInitializationHostService.cs

```diff
@@ -37,34 +33,23 @@ public async Task StartAsync(CancellationToken cancellationToken)
             _cancellationToken = cancellationToken;
 
             // The service should be registered in Linux Consumption only, but do additional check here.
-            if (_environment.IsLinuxConsumptionOnAtlas())
+            if (_environment.IsAnyLinuxConsumption())
             {
-                await ApplyStartContextIfPresent();
-            }
-            else if (_environment.IsLinuxConsumptionOnLegion())
-            {
-                _logger.LogInformation("Container has (re)started. Waiting for specialization");
+                await ApplyStartContextIfPresent(cancellationToken);
             }
         }
 
-        private async Task ApplyStartContextIfPresent()
+        private async Task ApplyStartContextIfPresent(CancellationToken cancellationToken)
         {
-            var startContext = await GetStartContextOrNullAsync();
+            (bool hasStartContext, string startContext) = await TryGetStartContextOrNullAsync(cancellationToken);
 
-            if (!string.IsNullOrEmpty(startContext))
+            if (hasStartContext && !string.IsNullOrEmpty(startContext))
```

Comment:
As there is already a check `string.IsNullOrEmpty(startContext)` - simplify `TryGetStartContextOrNullAsync` method signature  - return only `startContext` - better to avoid returning tuples

Reply:
Changed it earlier after suggestion from @yojagad 
Yogesh, what do you think?

Reply:
Let's change this to follow pattern similar to https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=net-7.0 --> return bool type with out value